client/src/components/Comparison/RelatedProducts/RelatedProducts.jsx

```diff
@@ -1,130 +1,56 @@
-    this.rightScrollerDisplay = this.rightScrollerDisplay.bind(this);
-  }
+  state = {
+    relatedProductIds: [],
+  };
 
+  // get the related products id when initially mounted
   componentDidMount() {
-    this.token = PubSub.subscribe('newProductId', (msg, data) => {
-      this.updateRelatedProducts(data.id)
-        .then(() => {
-          this.rightScrollerDisplay();
-        });
-    });
-
-    this.updateRelatedProducts(this.props.productId)
-      .then(() => {
-        this.rightScrollerDisplay();
-      });
+    const { productId } = this.props;
+    this.updateRelatedProducts(productId);
   }
 
-  componentWillUnmount() {
-    PubSub.unsubscribe(this.token);
-  }
-
-  handleScroll(isRight) {
-    return async () => {
-      const unitOffset = isRight ? -192 : 192;
-      const { leftOffset } = this.state;
-      await new Promise((resolve) => {
-        this.setState({
-          leftOffset: leftOffset + unitOffset,
-        }, resolve);
-      });
-      this.rightScrollerDisplay();
-    };
-  }
+  // when updated,
+  // if the productId changed, re-get the related products id
+  componentDidUpdate(prevProps) {
+    const { productId: prevId } = prevProps;
+    const { productId: currId } = this.props;
 
-  // if the right edge of the carousel is inside the container
-  // set this.state.isRightEdgeInBound to false
-  rightScrollerDisplay() {
-    const { clientWidth: conainerWidth } = this.main || {};
-    const { clientWidth: wrapperWidth } = this.carouselWrapper;
-    const { leftOffset } = this.state;
-    if (wrapperWidth + leftOffset <= conainerWidth) {
-      this.setState({
-        isRightEdgeInBound: true,
-      });
-    } else {
-      this.setState({
-        isRightEdgeInBound: false,
-      });
+    if (prevId !== currId) {
+      this.updateRelatedProducts(currId);
     }
   }
 
-  async updateRelatedProducts(productId) {
+  // get the related products id and update state
+  updateRelatedProducts = async (productId) => {
     try {
       const response = await instance.get(`/products/${productId}/related`);
       let { data } = response;
       data = [...new Set(data)];
 
-      await new Promise((res) => {
-        this.setState({
-          relatedProductIds: data,
-        }, res);
+      this.setState({
+        relatedProductIds: data,
       });
-    } catch (e) {
-      console.warn(e);
+    } catch (err) {
+      console.log(err);
     }
-  }
+  };
 
   render() {
-    const { leftOffset, isRightEdgeInBound, relatedProductIds } = this.state;
-    const { productId, updateProductId } = this.props;
+    const { relatedProductIds } = this.state;
+    const { updateProductId, productId } = this.props;
     return (
       <div className={relatedProducts['related-products']}>
-        <div className={relatedProducts.title} data-testid="title">
-          RELATED PRODUCTS
-        </div>
         <span>
           Current Product Id:
           {productId}
```

Comment:
@qzyan I think this should be tied to the `productId` in `App.jsx`. My expectation is that everytime the user clicks on a new product in `RelatedProducts` that the whole page rerenders for the new `productId`. Make sense?